pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -42,19 +42,22 @@ import (
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 	"github.com/openshift/origin/pkg/dockerregistry/server"
 	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
+	registryconfig "github.com/openshift/origin/pkg/dockerregistry/server/configuration"
+	"github.com/openshift/origin/pkg/dockerregistry/server/metrics"
 )
 
 // Execute runs the Docker registry.
 func Execute(configFile io.Reader) {
-	config, err := configuration.Parse(configFile)
+	dockerConfig, extraConfig, err := registryconfig.Parse(configFile)
 	if err != nil {
 		log.Fatalf("error parsing configuration file: %s", err)
 	}
-	setDefaultMiddleware(config)
-	setDefaultLogParameters(config)
+	setDefaultMiddleware(dockerConfig)
+	setDefaultLogParameters(dockerConfig)
 
 	ctx := context.Background()
-	ctx, err = configureLogging(ctx, config)
+	ctx = server.WithConfiguration(ctx, extraConfig)
```

Comment:
Nevertheless, I'd still want to avoid using the configuration from the context wherever possible. Let's use `server.AccessControllerParams` to configure the auth handler and pass the other configuration options directly to any other object we initialize here.